lesson21/main.js

```diff
@@ -119,23 +124,92 @@ const formingTableHeaderNameWithKey = (key) => {
 const createTableData = (usersData) => {
   const fragment = document.createDocumentFragment();
   usersData.forEach((userData) => {
-    const trOfTdata = document.createElement("tr");
+    const trOfTdata = createElementWithClassName("tr", "users-table__tr-td");
     /**
      *Extract value from each of usersData
      * @param {Object} userData the object of each users data
      * @param {String,Number} val value of each users data
      */
 
-    Object.values(userData).forEach((val) => {
+    Object.keys(userData).forEach((key) => {
+      if (key === "id") return;
       const td = createElementWithClassName("td", "users-table__td");
-      td.textContent = val;
+      td.textContent = userData[key];
       trOfTdata.appendChild(td);
     });
     fragment.appendChild(trOfTdata);
   });
   return fragment;
 };
 
+const createSortBtn = () => {
+  const btn = createElementWithClassName("button", "sort-btn");
+  const sortArrow = createElementWithClassName("img", "sort-img");
+  // bothのイメージをデフォルトとする
+  sortArrow.src = "./img/both.svg";
+  btn.appendChild(sortArrow);
+  return btn;
+};
+
+const renderSortBtn = () => {
+  const btn = createSortBtn();
+  const targetEle = document.getElementById("id");
+  targetEle.appendChild(btn);
+};
+
+const clickSortBtn = (usersData) => {
+  const sortArrow = document.querySelector(".sort-btn");
+  const sortArrowImg = document.querySelector(".sort-img");
+  sortArrow.addEventListener("click", () => {
+    changeSortStateAndArrowImg(sortArrowImg);
+    rerenderTableData(usersData);
+  });
+};
+
+const changeSortStateAndArrowImg = (sortArrow) => {
+  if (sortState === "both") {
+    sortState = "asc";
+    sortArrow.src = "/img/asc.svg";
+  } else if (sortState === "asc") {
+    sortState = "desc";
+    sortArrow.src = "/img/desc.svg";
+  } else {
+    sortState = "both";
+    sortArrow.src = "/img/both.svg";
+  }
```

Comment:
IMO
Maybe it would be easier to read if I made an early return here too. But that's my personal opinion!
```js
if (sortState === "both") {
    sortState = "asc";
    sortArrow.src = "/img/asc.svg";
    return;
  }
if (sortState === "asc") {
    sortState = "desc";
    sortArrow.src = "/img/desc.svg";
    return;
  }
sortState = "both";
sortArrow.src = "/img/both.svg";
```

Reply:
Thank you. I missed this part, now it looks more readable!
[b57dfb5](https://github.com/yuka830/js-lessons/pull/27/commits/b57dfb598f176fabec0cf0358e344503abe1e057)